cpp/arcticdb/processing/operation_types.hpp

```diff
@@ -373,18 +381,37 @@ bool operator()(int64_t t, uint64_t u) const {
 }
 };
 
-struct IsInOperator {
+struct MembershipOperator {
```

Comment:
Any particular reason why this needs to be a base class rather than just two templates?

Reply:
It started that way, but something didn't compile and it was late in the night so I went for this.

In hindsight, what didn't work was probably one of the function templates, not the class-level template, so I am happy to try again if you feel strongly about this?